internal/app/singularity/pull.go

```diff
@@ -177,71 +170,66 @@ func downloadImageCallback(totalSize int64, r io.Reader, w io.Writer) error {
 // OrasPull will download the image specified by the provided oci reference and store
 // it at the location specified by file, it will use credentials if supplied
 func OrasPull(name, ref string, force bool, ociAuth *ocitypes.DockerAuthConfig) error {
-	ref = strings.TrimPrefix(ref, "//")
-
-	spec, err := reference.Parse(ref)
-	if err != nil {
-		return fmt.Errorf("unable to parse oci reference: %s", err)
+	if !force {
+		if _, err := os.Stat(name); err == nil {
+			return fmt.Errorf("image file already exists: %q - will not overwrite", name)
+		}
 	}
 
-	// append default tag if no object exists
-	if spec.Object == "" {
-		spec.Object = SifDefaultTag
-		sylog.Infof("No tag or digest found, using default: %s", SifDefaultTag)
+	sysCtx := &ocitypes.SystemContext{
+		DockerAuthConfig: ociAuth,
 	}
 
-	credFn := func(_ string) (string, string, error) {
-		if ociAuth != nil {
-			return ociAuth.Username, ociAuth.Password, nil
-		}
-
-		return "", "", nil
+	sum, err := oras.ImageSHA(ref, sysCtx)
+	if err != nil {
+		return fmt.Errorf("failed to get checksum for %s: %s", ref, err)
 	}
-	resolver := docker.NewResolver(docker.ResolverOptions{Credentials: credFn})
 
-	wd, err := os.Getwd()
+	imageName := uri.GetName("oras:" + ref)
+
+	imagePath := cache.OrasImage(sum, imageName)
```

Comment:
_I_ would call this cacheFilename or cachedImage or some variation of that, to underline that this is the cached file, not the image file that the user is asking for.